tests/PKSim.Tests/Presentation/EditTimeProfileAnalysisChartPresenterSpecs.cs

```diff
@@ -258,7 +266,100 @@ protected override void Context()
+         A.CallTo(() => _representationInfoRepository.DisplayNameFor(_percentileStatisticalAggregation)).Returns(_percentileId);
+         sut = new TestEditTimeProfileAnalysisChartPresenter(_view, _timeProfilerChartPresenter, _timeProfileChartDataCreator,
+            _populationSimulationAnalysisStarter, _populationAnalysisTask, _colorGenerator, _observedDataTask, _pkAnalysisPresenter, _dimensionRepository, _presenterSettingsTask, _representationInfoRepository, _statisticalDataCalculator, _pkAnalysesTask);
+
+         var model = A.Fake<IModel>();
+         A.CallTo(() => model.MoleculeNameFor("Organism|PeripheralVenousBlood|Esomeprazole|Plasma (Peripheral Venous Blood)")).Returns("Esomeprazole");
+         A.CallTo(() => model.MoleculeNameFor("Organism|PeripheralVenousBlood|Esomeprazole-2|Plasma (Peripheral Venous Blood)")).Returns("Esomeprazole-2");
+         _simulation = A.Fake<PopulationSimulation>();
+         A.CallTo(() => _simulation.Model).Returns(model);
+         A.CallTo(() => _simulation.Compounds).Returns(new[] { new Compound() { Name = "Esomeprazole" }, new Compound() { Name = "Esomeprazole-2" } });
+         _simulation.AddCompoundPK(new CompoundPK() { CompoundName = "Esomeprazol-2" });
+         var analysis = new TimeProfileAnalysisChart();
+         analysis.PopulationAnalysis = _populationStatisticalAnalysis;
+         sut.InitializeAnalysis(analysis, _simulation);
+      }
+
+      protected override void Because()
+      {
+         var pkParameters = new[]
+         {
+            new TestQuantityPKParameter() { Name = "Name 1", QuantityPath = "Organism|PeripheralVenousBlood|Esomeprazole|Plasma (Peripheral Venous Blood)",   _values = new[] { 0.000f, 0.050f, 0.025f, 0.075f, 1.000f } },
+            new TestQuantityPKParameter() { Name = "Name 2", QuantityPath = "Organism|PeripheralVenousBlood|Esomeprazole|Plasma (Peripheral Venous Blood)",   _values = new[] { 0.00f,  0.25f,  0.75f,  0.50f,  1.00f  } },
+            new TestQuantityPKParameter() { Name = "Name 3", QuantityPath = "Organism|PeripheralVenousBlood|Esomeprazole|Plasma (Peripheral Venous Blood)",   _values = new[] { 0.0f,   2.5f,   5.0f,   7.5f,   10.0f  } },
+            new TestQuantityPKParameter() { Name = "Name 1", QuantityPath = "Organism|PeripheralVenousBlood|Esomeprazole-2|Plasma (Peripheral Venous Blood)", _values = new[] { 0f,     0f,     0f,     0f,     0f } },
+            new TestQuantityPKParameter() { Name = "Name 2", QuantityPath = "Organism|PeripheralVenousBlood|Esomeprazole-2|Plasma (Peripheral Venous Blood)", _values = new[] { 0f,     0f,     0f,     0f,     0f } },
+            new TestQuantityPKParameter() { Name = "Name 3", QuantityPath = "Organism|PeripheralVenousBlood|Esomeprazole-2|Plasma (Peripheral Venous Blood)", _values = new[] { 0f,     0f,     0f,     0f,     0f } }
+         };
+
+         _pkAnalyses = (sut as TestEditTimeProfileAnalysisChartPresenter).AggregatePKAnalysis(_simulation, pkParameters, "Esomeprazole");
```

Comment:
you would not need to create this presenter with all those fakes if the logic was outside of presneetaiton lgoci